File: servico-musica-java/java.rest/src/main/java/servico/java/rest/controller/PlaylistController.java
package servico.java.rest.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import servico.java.rest.entity.Musica;
import servico.java.rest.entity.Playlist;
import servico.java.rest.entity.Usuario;
import servico.java.rest.repository.MusicaRepository;
import servico.java.rest.repository.PlaylistRepository;
import servico.java.rest.repository.UsuarioRepository;

import java.util.Collections;
import java.util.List;
import java.util.Optional;

@RestController
@RequestMapping("/servico-rest-java/playlist")
public class PlaylistController {

    @Autowired
    private PlaylistRepository playlistRepository;

    @Autowired
    private UsuarioRepository usuarioRepository;

    @Autowired
    private MusicaRepository musicaRepository;

    @PostMapping("/usuario/{idUsuario}")
    public List<Playlist> playlistUsuario(@PathVariable long idUsuario){

        Optional<Usuario> usuario = usuarioRepository.findById(idUsuario);

        if (usuario.isEmpty()){
            return Collections.emptyList();
        }

        return usuario.get().getPlaylistsUsuario();
    }

    @PostMapping("/musica/{idMusica}")
    public List<Playlist> musicasEspecifica(@PathVariable long idMusica){

        return playlistRepository.findByMusicasId(idMusica);
    }


}

File: servico-musica-kotlin/kotlin.rest/src/main/kotlin/servico/kotlin/rest/controller/PlaylistController.kt
package servico.kotlin.rest.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import servico.kotlin.rest.entity.Playlist
import servico.kotlin.rest.repository.MusicaRepository
import servico.kotlin.rest.repository.PlaylistRepository
import servico.kotlin.rest.repository.UsuarioRepository

@RestController
@RequestMapping("/servico-rest-kotlin/playlist")
class PlaylistController {
    @Autowired
    private val playlistRepository: PlaylistRepository? = null

    @Autowired
    private val usuarioRepository: UsuarioRepository? = null

    @Autowired
    private val musicaRepository: MusicaRepository? = null

    @PostMapping("/usuario/{idUsuario}")
    fun playlistUsuario(@PathVariable idUsuario: Long): MutableList<Playlist?> {
        val usuario = usuarioRepository!!.findById(idUsuario)

        if (usuario.isEmpty()) {
            return mutableListOf<Playlist?>()
        }

        return usuario.get().playlistsUsuario
    }

    @PostMapping("/musica/{idMusica}")
    fun musicasEspecifica(@PathVariable idMusica: Long): MutableList<Playlist?>? {
        return playlistRepository!!.findByMusicasId(idMusica)
    }
}